### ISSUE TITLE
deprecate asClass 

### ISSUE THREAD

**Issue by Ducasse** [guillep/pharo-core#14970]
asClass is a bad idea and we should remove it from the system.

**Comment by MarcusDenker** [comment]
What is interesting is that compiling an Undeclared now creates very similar code: push the name, send a message.

```
 41 <20> pushConstant: #MyClass
42 <81> send: asClass
```

```
41 <20> pushConstant: ##MyClass
42 <81> send: runtimeUndeclaredRead
```

The constant pushed in the second case is a UndeclaredVariable instance (which knows it's name). 

for now #runtimeUndeclaredRead raises an exception when the var is still Undeclared, if a definition is loaded it gets repaired automatically (the Undeclared Variable is replaced by a GlobalVariable).

Right now we try to avoid Undeclared at any cost, this is why #asClass was added. But the "fear of the Undeclared" is just something we decided. We could relax it and see Undeclared vars as nothing evil in case they are used for pluggability. 

(I just add it here as part of taking notes/thinking about Undeclared, see issue https://github.com/pharo-project/pharo/issues/14871)
